### PR TITLE
Indoor maps now display again in DirectionsActivity

### PR DESCRIPTION
**Pull request by felixlapierre**
When #210 was merged the OnZoomListener was modified in such a way that it broke the display of indoor maps in the directions flow. This pull request is a quick fix to make the maps display again. Further refactoring should be considered to make the whole thing more extensible.

### CODE REVIEW

File: app/src/main/java/com/example/campusguide/DirectionsActivity.kt [hunk 1]
```diff
@@ -63,7 +63,16 @@ class DirectionsActivity : AppCompatActivity(), AdapterView.OnItemClickListener
...
+        //TODO: Refactor GoogleMapInitializer so it has less nullable constructor properties
+        val initializer = GoogleMapInitializer(
+            this,
+            map,
+            "directions_activity_map",
+            null,
+            null,
+            null,
+            BuildingIndexSingleton.getInstance(assets)
+        )
```
Comment (MikeBrichko) [inline app/src/main/java/com/example/campusguide/DirectionsActivity.kt:75]: What is the reason for all of the null values in the constructor?

**Review by MikeBrichko — APPROVED** [review]

**Review by jun-loke — CHANGES_REQUESTED** [review]
Lint then I can Approve

**Review by jun-loke — APPROVED** [review]
All good

File: app/src/main/java/com/example/campusguide/DirectionsActivity.kt (hunk 1 above)
Comment (felixlapierre) [reply app/src/main/java/com/example/campusguide/DirectionsActivity.kt:75]: Answer:
GoogleMapsInitializer has a large amount of optional parameters that determine which features are enabled. For example, you can add a polygon click listener which is necessary for popups when clicking on Concordia buildings. However, in activities where you don't want those features, you can just pass null.

The BuildingIndexSingleton is one of those optional parameters, it just happens to be the last one in the list. So I need to pass null for all the previous optional parameters.

This is clearly a big code smell, so I added a TODO which you can see a few lines higher to represent that piece of technical debt which I do not have time to address right now.

**Pull request merged**